Minor compilation tweaks for the LLVM wasm backend

sunfishcode opened the pull request.

dschuff commented:
LGTM. How big of a win in compile time is this?
Perhaps we eventually want do do this in wasm's clang too.

kripken commented on tools/shared.py:
```diff
@@ -884,9 +884,12 @@ def get_llvm_target():
   COMPILER_OPTS = COMPILER_OPTS + ['-DEMSCRIPTEN',
-                                   '-D__EMSCRIPTEN__']
+                                   '-D__EMSCRIPTEN__',
+                                   '-D_LIBCPP_ABI_VERSION=2']
```
why is this just for the wasm target?

sunfishcode replied:
It's an ABI-breaking change, and I didn't want to risk breakage for asm.js. However, if you think asm.js is ok with a C++ ABI break, then sure, we can do it too.

kripken replied:
The only risk is if someone builds some code as separate modules with different versions of emscripten? We don't really make guarantees about that, our linking has changed quite a bit over time. So I think it's fine to do this change for everything.

juj replied:
Also fine for me. We have stressed to developers that .bc files are not compatible across Emscripten versions and that there is no locked in ABI across versions, so we should be free to do nimble stuff like this, where it helps. It is very good to note known breakages however, since these are good type of changes to document then in ChangeLog.markdown when I collect the tag updates there.

sunfishcode commented:
Updated the patch to set the new ABI for asm.js too.

@juj I added a blurb to ChangeLog.markdown.

@dschuff The combiner-aa flag change is just a few percent compile-time speedup.

kripken commented:
Looks good to me, with

1. Bumping of the version number (to note the abi change; also will force a rebuild of bitcode for that)
2. Do some testing before landing, unless this is an ABI change you are certain will not break anything? I would run `other` and maybe `random100` from the main suite.

After this lands we'll also need to update the version in the other 2 repos and update tags, I can do that.

sunfishcode commented:
other and random100 pass, except the closure, scons, etc. tests. Version number bumped.

kripken commented:
Thanks, merged and version bumped.
